presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -74,31 +74,59 @@ class PartitionAndSerializeOperator : public Operator {
 
   void addInput(RowVectorPtr input) override {
     input_ = std::move(input);
-  }
+    const auto numInput = input_->size();
 
-  RowVectorPtr getOutput() override {
-    if (!input_) {
-      return nullptr;
-    }
+    // Reset state variables.
+    processUntil_ = processFrom_ = 0;
+    rowSizes_.clear();
+    rowSizes_.resize(numInput);
 
-    const auto numInput = input_->size();
+    compactRow_ =
+        std::make_unique<velox::row::CompactRow>(reorderInputsIfNeeded());
+    calculateRowSize();
 
-    // TODO Reuse output vector.
-    auto output = BaseVector::create<RowVector>(outputType_, numInput, pool());
-    auto partitionVector = output->childAt(0)->asFlatVector<int32_t>();
+    // Process partitionVector and replicateVector once, and reuse on subsequent
+    // batch.
+    backingOutputVector_ = BaseVector::create<RowVector>(
```

Comment:
Can we create backingOutputVector_ once? Rename it to outputVector_? See output_ example in HashProbe? Thanks!